### ISSUE TITLE
JDK19 serviceability_jvmti_j9_0_FAILED serviceability/jvmti/vthread/RawMonitorTest/RawMonitorTest.java Fatal Error in RawMonitorWait

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#16753]
Failure link
------------

From [an internal build](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk19_j9_extended.openjdk_x86-64_mac/39/)(`mac11x64rt10`):
```
openjdk version "19.0.2" 2023-01-17
IBM Semeru Runtime Open Edition 19.0.2+7 (build 19.0.2+7)
Eclipse OpenJ9 VM 19.0.2+7 (build master-042864b42, JRE 19 Mac OS X amd64-64-Bit Compressed References 20230117_46 (JIT enabled, AOT enabled)
OpenJ9   - 042864b42
OMR      - b86560818
JCL      - edc950934d based on jdk-19.0.2+7)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=customized&amp;TARGET=serviceability_jvmti_j9_0&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=0&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=adoptium%3Amaster&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=x86-64_mac&amp;GENERATE_JOBS=true&amp;KEEP_REPORTDIR=true&amp;PERSONAL_BUILD=false&amp;DOCKER_REGISTRY_DIR=&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;RERUN_ITERATIONS=3&amp;SETUP_JCK_RUN=false&amp;DOCKER_REGISTRY_URL_CREDENTIAL_ID=&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk19u%2Fjdk19u-mac-x64-openj9%2F46%2Fibm-semeru-open-jdk_x64_mac_JDK19U_2023-02-18-17-00.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk19u%2Fjdk19u-mac-x64-openj9%2F46%2Fibm-semeru-open-debugimage_x64_mac_JDK19U_2023-02-18-17-00.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk19u%2Fjdk19u-mac-x64-openj9%2F46%2Fibm-semeru-open-testimage_x64_mac_JDK19U_2023-02-18-17-00.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk19u%2Fjdk19u-mac-x64-openj9%2F46%2Fibm-semeru-open-jre_x64_mac_JDK19U_2023-02-18-17-00.tar.gz&amp;BUILD_IDENTIFIER=&amp;JENKINS_KEY=&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=false&amp;USE_JRE=false&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=83181e25-eea4-4f55-8b3e-e79615733226&amp;JDK_VERSION=19&amp;DOCKER_REGISTRY_URL=&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=https%3A%2F%2Fgithub.com%2Fibmruntimes%2Fopenj9-openjdk-jdk19&amp;JCK_GIT_BRANCH=main&amp;RELEASE_TAG=&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=&amp;CLOUD_PROVIDER=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=openj9&amp;LABEL_ADDITION=%21sw.os.osx.10_11&amp;ARTIFACTORY_REPO=&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=&amp;DOCKERIMAGE_TAG=&amp;JDK_IMPL=openj9&amp;TEST_TIME=&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=%23rt-jenkins&amp;DYNAMIC_COMPILE=false&amp;RELATED_NODES=&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;APPLICATION_OPTIONS=&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=4e18ffe7-b1b1-4272-9979-99769b68bcc2&amp;ARCHIVE_TEST_RESULTS=false&amp;NUM_MACHINES=&amp;OPENJDK_SHA=&amp;TRSS_URL=http%3A%2F%2Ftrss1.fyre.ibm.com&amp;USE_TESTENV_PROPERTIES=false&amp;BUILD_LIST=openjdk&amp;UPSTREAM_JOB_NUMBER=&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TIME_LIMIT=20&amp;JVM_OPTIONS=&amp;PARALLEL=None) - Change TARGET to run only the failed test targets.

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
[2023-02-18T19:34:32.237Z] variation: Mode150
[2023-02-18T19:34:32.237Z] JVM_OPTIONS:  -XX:+UseCompressedOops 

[2023-02-18T19:36:33.674Z] TEST: serviceability/jvmti/vthread/RawMonitorTest/RawMonitorTest.java

[2023-02-18T19:36:33.691Z] VT2/ForkJoinPool-1-worker-3: rawMonitorEnter: entered
[2023-02-18T19:36:33.691Z] VT2/ForkJoinPool-1-worker-4: rawMonitorWait: before waiting
[2023-02-18T19:36:33.691Z] check_jvmti_status: JVMTI function returned error: JVMTI_ERROR_NOT_MONITOR_OWNER (51)
[2023-02-18T19:36:33.691Z] STDERR:
[2023-02-18T19:36:33.691Z] 
[2023-02-18T19:36:33.691Z] Fatal error: Fatal Error in RawMonitorWait

[2023-02-18T19:36:38.788Z] serviceability_jvmti_j9_0_FAILED
```

[50x internal grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/31176/) - 7/50 `RawMonitorTest` failures

**Comment by pshipton** [comment]
2 of the 5 grinders failed RawMonitorTest.java
Failed 4/10 times on osxrt12
Failed 1/10 time on mac11x64rt10

@JasonFengJ9 there is also another failure in serviceability/jvmti/vthread/BreakpointInYieldTest/BreakpointInYieldTest.java that doesn't have an issue yet.

**Comment by JasonFengJ9** [comment]
Also seen on [JDK19 AIX grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder_iteration_4/1639/consoleFull)
```
17:46:48  openjdk version "19.0.2" 2023-01-17
17:46:48  IBM Semeru Runtime Open Edition 19.0.2+7 (build 19.0.2+7)
17:46:48  Eclipse OpenJ9 VM 19.0.2+7 (build master-042864b42, JRE 19 AIX ppc64-64-Bit Compressed References 20230117_46 (JIT enabled, AOT enabled)
17:46:48  OpenJ9   - 042864b42
17:46:48  OMR      - b86560818
17:46:48  JCL      - edc950934d based on jdk-19.0.2+7)
```

**Comment by tajila** [comment]
@babsingh Please take a look

**Comment by babsingh** [comment]
The fix (https://github.com/eclipse-openj9/openj9/pull/16790) was merged yesterday. @JasonFengJ9 Can you please verify the fix using a grinder and close this issue?

**Comment by JasonFengJ9** [comment]
@babsingh The fix missed the last nightly build, we can try tonight's build later.
```
openjdk version "19.0.2-beta" 2023-01-17
IBM Semeru Runtime Open Edition 19.0.2+7-202303010131 (build 19.0.2-beta+7-202303010131)
Eclipse OpenJ9 VM 19.0.2+7-202303010131 (build master-049495700, JRE 19 Mac OS X amd64-64-Bit Compressed References 20230228_53 (JIT enabled, AOT enabled)
OpenJ9   - 049495700
OMR      - 084e87a92
JCL      - 3c83ead7fc based on jdk-19.0.2+7)

645d272500bc0044bab6a41c48f490036b561a42 Merge pull request #16790 from babsingh/pin_vt_jvmti_monitor_funcs
04949570046359b6d9a5a21cd06d13a4e66434be Merge pull request #16771 from RSalman/criu-opts-v2 --> latest nightly
```

**Comment by JasonFengJ9** [comment]
[a 50x grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/31337/console)
```
09:19:38  openjdk version "19.0.2-beta" 2023-01-17
09:19:38  IBM Semeru Runtime Open Edition 19.0.2+7-202303020130 (build 19.0.2-beta+7-202303020130)
09:19:38  Eclipse OpenJ9 VM 19.0.2+7-202303020130 (build master-6cddd7614, JRE 19 Mac OS X amd64-64-Bit Compressed References 20230301_54 (JIT enabled, AOT enabled)
09:19:38  OpenJ9   - 6cddd7614
09:19:38  OMR      - 53b3b1bff
09:19:38  JCL      - 3c83ead7fc based on jdk-19.0.2+7)
```
`Fatal Error in RawMonitorWait` wasn't reproduced. There were timeout, and another known failure
* https://github.com/eclipse-openj9/openj9/issues/16756

Closing

**Comment by JasonFengJ9** [comment]
This failure appeared at [JDK19 milestone 1 build](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk19_j9_extended.openjdk_ppc64le_linux/41/)
@babsingh [v0.38.0-release](https://github.com/eclipse-openj9/openj9/tree/v0.38.0-release) (JDK19 release) needs this fix as well.

**Comment by babsingh** [comment]
- For Java 19, the fix was delivered in the 0.37 branch today (8 hrs ago): https://github.com/eclipse-openj9/openj9/pull/16810.
- Reason for the failure: [JDK19 milestone 1 build](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk19_j9_extended.openjdk_ppc64le_linux/41/) was run 22 hrs ago, before the fix was merged in the 0.37 branch.
- 0.38 branch does not need the fix since it will only be used to deliver Java 8, 11 and 17.

**Comment by babsingh** [comment]
@JasonFengJ9 Please close the issue if it is sufficient to deliver the fix only in the 0.37 branch. If not, let me know, and I will deliver the fix to the 0.38 branch.

**Comment by JasonFengJ9** [comment]
Agreed, next JDK19 build should pick up the fix, and 0.38 is not for JDK19 delivery.

Closing
